share/pot/start.sh

```diff
@@ -427,6 +442,29 @@ _js_start()
 	else
 		_param="$_param nopersist"
 	fi
+	if [ "$(_get_conf_var "$_pname" "pot.attr.no-rc-script")" = "YES" ]; then
+		if [ "$( _get_pot_network_stack "$_pname" )" = "ipv4" ]; then
+			prec=100
+		else
+			prec=35
+		fi
+		cat >>"${POT_FS_ROOT}/jails/$_pname/m/tmp/tinirc" <<-EOT
+		if sysctl -n kern.features.inet6 >/dev/null 2>&1; then
+		        ip6addrctl flush >/dev/null 2>&1
+		        ip6addrctl install /dev/stdin <<EOF
+		        ::1/128		 50	 0
+		        ::/0		 40	 1
+		        ::ffff:0:0/96	 $prec	 4
+		        2002::/16	 30	 2
+		        2001::/32	  5	 5
+		        fc00::/7	  3	13
+		        ::/96		  1	 3
+		        fec0::/10	  1	11
+		        3ffe::/16	  1	12
+		EOF
+		fi
+		EOT
+	fi
```

Comment:
This is the change for ip6addr policy (all the rest is adding missing epair checks). Let me know if this should also be divided. In general I think this is relatively clean solution.

Reply:
the solution is clean, but there is one thing I'm unsure of: with `ipv4` stack or `ipv6` stack, does it make sense to have address selection policy? shouldn't it matters on `dual` only?

Reply:
@pizzamig It is especially important on a non-dual _pot_ stack, specifically with ipv4, since `pot.stack=ipv4` is only about what pot configures in the jail, not about the OS/the vnet stack supporting IPv6.

# Example

    # pot create -p box -b 13.0 -t single -N public-bridge -S ipv4 && \
    pot set-attr -p box -A no-rc-script -V YES && \
    pot set-attr -p box -A persistent -V NO  && \
    pot set-cmd -p box -c '/sbin/ping -c3 www.google.com'
    # pot start box
    ===>  Starting the pot box
    add net default: gateway 10.192.0.1
    ping: UDP connect: No route to host 

    # pot set-cmd -p box \
    -c 'sh -c "service ip6addrctl start && /sbin/ping -c3 www.google.com"'
    # pot start box
    ===>  Starting the pot box
    add net default: gateway 10.192.0.1
    PING www.google.com (142.251.36.196): 56 data bytes
    64 bytes from 142.251.36.196: icmp_seq=0 ttl=118 time=15.412 ms
    64 bytes from 142.251.36.196: icmp_seq=1 ttl=118 time=13.560 ms
    64 bytes from 142.251.36.196: icmp_seq=2 ttl=118 time=14.952 ms
    # pot destroy -p box


# Explanation

What happens under the hood is:
- DNS lookup of www.google.com
- Resolver sees both A and AAAA records
- As the kernel supports IPv6, it checks the address selection policy
- Based on that, ping decides to ping www.google.com over IPv6
- Ping realizes that there is no route to www.google.com's IPv6 address (as we only have IPv6 on localhost, but no gateway for IPv6) and reports `UDP connect: No route to host `

Looking at interfaces inside the jail shows `lo0` holding `::1`:

    # pot create -p box -b 13.0 -t single -N public-bridge -S ipv4 && \
    pot set-attr -p box -A no-rc-script -V YES && \
    pot set-attr -p box -A persistent -V NO  && \
    pot set-cmd -p box -c '/sbin/ifconfig'
    # pot start box
    ===>  Starting the pot box
    add net default: gateway 10.192.0.1
    lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> metric 0 mtu 16384
    	options=680003<RXCSUM,TXCSUM,LINKSTATE,RXCSUM_IPV6,TXCSUM_IPV6>
    	inet 127.0.0.1 netmask 0xff000000
    	inet6 ::1 prefixlen 128
    	inet6 fe80::1%lo0 prefixlen 64 scopeid 0x1
    	groups: lo
    	nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
    epair8b: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
    	options=8<VLAN_MTU>
    	ether ...
    	inet 10.192.0.4 netmask 0xffc00000 broadcast 10.255.255.255
    	groups: epair
    	media: Ethernet 10Gbase-T (10Gbase-T <full-duplex>)
    	status: active
    	nd6 options=29<PERFORMNUD,IFDISABLED,AUTO_LINKLOCAL>
    # pot destroy -p box

# Alternatives

Technically, there might be ways to somehow really disable IPv6 in the VNET stack of the pot, but I would recommend against that, as I don't think this is well tested and a lot of software expects to have an IPv6 stack available these days anyway.

Having both stacks enabled and configuring the IPv6 address selection policy is what happens on non-tinirc FreeBSD on every boot, that's why this was implemented here.

Reply:
I totally got your point.
In my CI in the cloud, I set up ipv4 only and never encountered this issue. Happy to keep it as you proposed